models/demos/tg/llama3_70b/tt/llama_generation_galaxy.py

```diff
@@ -58,16 +56,10 @@ def forward(self, tokens: torch.Tensor, start_pos: int):
             return self.decode_forward(tokens, start_pos)
 
     def decode_forward(self, tokens: torch.Tensor, start_pos: int):
-        self._update_model_config("decode", tokens.shape[0], 1)
```

Comment:
Why do we remove this? This is super useful for switch between prefill and decode mode due to different program_configs/compute_kernel_configs, etc. I see 2 solutions for this: 
1) Keep update_model_config whenever we call forward method.
2) Update model components to contain both configs for prefill and decode at the same time and use according to llm_mode.
Let me know what you think!

Reply:
Thanks for reviewing, Djordje!

In TG, the only difference between prefill and decode in the model_config was `LLM_MODE`, which I removed. Now, every module is passed a `mode` parameter which determines whether to run decode or prefill forwards.

One unfortunate side effect is that now all modules recreate their program configs on every forward pass, based on the `mode`. I recommend modifying the module code such that prefill and decode program configs are saved once on instantiation. Then we can remove the `self.get_model_config(mode)` calls from every module's forward.